File: internal/models/task.go
package models

import (
	"time"

	"github.com/google/uuid"
)

type Task struct {
	ID          uuid.UUID  `json:"id"`
	Title       string     `json:"title"`
	Description string     `json:"description"`
	Status      string     `json:"status"`
	Assignee    string     `json:"assignee"`
	DueDate     *time.Time `json:"due_date"`
	Version     int        `json:"version"`
	CreatedAt   time.Time  `json:"created_at"`
	UpdatedAt   time.Time  `json:"updated_at"`
	DeletedAt   *time.Time `json:"deleted_at,omitempty"`
}

type TaskHistory struct {
	ID        int64     `json:"id"`
	TaskID    uuid.UUID `json:"task_id"`
	FieldName string    `json:"field_name"`
	OldValue  *string   `json:"old_value"`
	NewValue  *string   `json:"new_value"`
	ChangedAt time.Time `json:"changed_at"`
}

type TaskFilter struct {
	Status      string    `form:"status"`
	Assignee    string    `form:"assignee"`
	DueDateFrom time.Time `form:"due_date_from"`
	DueDateTo   time.Time `form:"due_date_to"`
	Offset      int       `form:"offset,default=0"`
	Limit       int       `form:"limit,default=20"`
}

File: frontend/src/types/task.ts
export interface Task {
    id: string;
    title: string;
    description: string | null;
    status: 'new' | 'in_progress' | 'done' | 'cancelled';
    assignee: string | null;
    due_date: string | null;
    version: number;
    created_at: string;
    updated_at: string;
    deleted_at: string | null;
  }
  
  export type CreateTaskData = Omit<Task, 'id' | 'version' | 'created_at' | 'updated_at' | 'deleted_at'> & {
    due_date?: string | null;
  };
  
  export type UpdateTaskData = Partial<Omit<Task, 'id' | 'created_at' | 'updated_at' | 'deleted_at'>> & {
    version: number;
  };
  
  export interface TaskFilter {
    status?: string;
    assignee?: string;
    due_date_from?: string;
    due_date_to?: string;
    offset?: number;
    limit?: number;
  }